helm-chart/todoapp/templates/deployment.yml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.Release.Name}}-{{.Chart.Name}}
+  namespace: {{.Values.todoapp.namespace}}
+spec:
+  replicas: 1
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.todoapp.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.todoapp.rollingUpdate.maxUnavailable}}
+  selector:
+    matchLabels:
+      app: {{.Chart.Name}}
+  template:
+    metadata:
+      labels:
+        app: {{.Chart.Name}}
+    spec:
+      serviceAccountName: {{ .Values.todoapp.RBAC.name}}
+      containers:
+      - name: {{.Values.todoapp.image.name}}
+        image: {{.Values.todoapp.image.repository}}:{{.Values.todoapp.image.version}}
+        volumeMounts:
+        - name: data
+          mountPath: /app/data
+        - name: app-secrets-volume
+          mountPath: "/app/secrets"
+          readOnly: true
+        - name: app-config-volume
+          mountPath: "/app/configs"
+          readOnly: true
+        resources:
+          requests:
+            memory: {{.Values.todoapp.resources.requests.memory}}
+            cpu: {{.Values.todoapp.resources.requests.cpu}}
+          limits:
+            memory: {{.Values.todoapp.resources.limits.memory}}
+            cpu: {{.Values.todoapp.resources.limits.cpu}}
+        env:
+        - name: PYTHONUNBUFFERED
+          valueFrom:
+            configMapKeyRef:
+              name: {{ .Chart.Name }}-config
+              key: PYTHONUNBUFFERED
+        {{- range $key, $value := .Values.todoapp.secret }}
+        - name: {{ $key }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ $.Release.Name }}-secret
+              key: {{ $key }}
+        {{ end }}
```

Comment:
Make sure that the `todoapp.secret` values in your `values.yaml` file are properly defined. These values should match the keys in the secret and be correctly base64 encoded.